nixos/modules/services/networking/yggdrasil.xml

```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0"?>
+<chapter xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xi="http://www.w3.org/2001/XInclude" version="5.0" xml:id="module-services-networking-yggdrasil">
+  <title>Yggdrasil</title>
+  <para>
+    <emphasis>Source:</emphasis>
+    <filename>modules/services/networking/yggdrasil/default.nix</filename>
+  </para>
+  <para>
+    <emphasis>Upstream documentation:</emphasis>
+    <link xlink:href="https://yggdrasil-network.github.io/"/>
+  </para>
+  <para>
+Yggdrasil is an early-stage implementation of a fully end-to-end encrypted
+IPv6 network. It is lightweight, self-arranging, supported on multiple
+platforms and allows pretty much any IPv6-capable application to communicate
+securely with other Yggdrasil nodes.
+</para>
+  <para>
+When enabled the Yggdrasil service creates a tun device with a unique IPv6 in
+the <literal>200::/8</literal> address space. This device and address can be
+used to communicate with all other connected nodes in the
+<literal>200::/7</literal> Yggdrasil network. Each Yggdrasil node can also
+route a subnet using the first half of its address with the eighth bit masked
+as a prefix, that is to say, within the <literal>300::/8</literal> space. This
+prefix may be announced to locally attached networks.
+</para>
+  <para>
+Despite the normal distribution of nodes within the <literal>200::/8</literal>
+space, the addresses of connected nodes can be readily enumerated, unlike the
+the traditional global IPv6 network. For this reason, all open services on a
+node should be considered public, or the system firewall should be enabled.
+</para>
+  <section xml:id="module-services-networking-yggdrasil-configuration">
+    <title>Configuration</title>
+    <section xml:id="module-services-networking-yggdrasil-configuration-simple">
+      <title>Simple ephemeral node</title>
+      <para>
+An annotated example of a simple configuration:
+<programlisting>
+{
+  services.yggdrasil = {
+    enable = true;
+    config = {
+      Peers = [
+        # Yggdrasil will automatically connect and "peer" with other nodes it
+        # discovers via link-local multicast annoucements. Unless this is the
+        # case (it probably isn't) a node needs peers within the existing
+        # network that it can tunnel to.
+        "tcp://1.2.3.4:1024"
+        "tcp://1.2.3.5:1024"
+      ];
+    };
+  };
+}
+</programlisting>
+   </para>
+    </section>
+    <section xml:id="module-services-networking-yggdrasil-configuration-prefix">
+      <title>Persistent node with prefix</title>
+      <para>
+A node with a fixed address that announces a prefix:
+<programlisting>
+let
+  address = "210:5217:69c0:9afc:1b95:b9f:8718:c3d2";
+  prefix = "310:5217:69c0:9afc";
+  # taken from the output of "yggdrasilctl getself".
```

Comment:
These are great configuration examples. I think it would be helpful to explain that in order for `yggdrasilctl getself` to work, the `yggdrasil` service needs to be running with `persistentKeys = true`.  I suggest breaking this into two sections, the first to tell the user how to make a node with a persistent address and how to find out what that address is, and the second to demonstrate how to make the node advertise the address.

Reply:
I made a note about `persistentKeys` above in the first example so the second doesn't get too heavy. Still not much about using yggdrasilctl.